src/compiler/checker.ts

```diff
@@ -6097,16 +6103,46 @@ namespace ts {
             return symbol ? getLocalTypeParametersOfClassOrInterfaceOrTypeAlias(symbol) : undefined;
         }
 
+        function filterNulableTypes(union: UnionType): Type[] {

```

Comment:
Spelling. Not sure you need to keep this function, but if you do, have it operate on any kind of type in the same way as `mapType` and `filterType`.